pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/inv/OffHeapBitmapInvertedIndexCreator.java

```diff
@@ -60,196 +56,151 @@
  * </ul>
  */
 public final class OffHeapBitmapInvertedIndexCreator implements InvertedIndexCreator {
-  // Maximum number of values due to 2GB file size limit
-  public static final int MAX_NUM_VALUES = Integer.MAX_VALUE / V1Constants.Numbers.INTEGER_SIZE;
-
-  private final String _columnName;
+  private final File _invertedIndexFile;
   private final boolean _singleValue;
   private final int _cardinality;
   private final int _numDocs;
   private final int _numValues;
-  private final File _indexDir;
-  private final File _invertedIndexFile;
 
-  private ByteBuffer _valueByteBuffer;
-  private IntBuffer _valueIntBuffer;
-
-  // For multi-value column only
-  private int _valueBufferOffset;
-  private ByteBuffer _numValuesByteBuffer;
-  private IntBuffer _numValuesIntBuffer;
-
-  // Posting list related buffers
-  // Constructed in first pass
-  private ByteBuffer _postingListLengthByteBuffer;
-  private IntBuffer _postingListLengthIntBuffer;
-  // Constructed in second pass
-  private ByteBuffer _postingListValueByteBuffer;
-  private ByteBuffer _postingListCurrentOffsetByteBuffer;
+  // For single-valued and multi-valued column
+  private int _nextDocId;
+  private LBuffer _valueBuffer;
+  private LBuffer _postingListIndexBuffer;
```

Comment:
Its not clear what "postingList" refers to. Could you add clarifying comments?

Reply:
Will add more comments